contracts/SnappAuction.sol

```diff
@@ -187,6 +200,48 @@ contract SnappAuction is SnappBase {
         }
     }
 
+    function auctionSolutionBid(
+        uint slot,
+        bytes32 _currStateRoot,
+        bytes32 proposedStateRoot,
+        uint proposedObjectiveValue
+    ) public {
+        require(
+            slot == 0 || auctions[slot-1].appliedAccountStateIndex != 0,
+            "Previous auction not yet resolved!"
+        );
+        require(coreData.stateRoots[stateIndex()] == _currStateRoot, "Incorrect state root");
+
+        // Ensure that auction batch is inactive, unprocessed and in correct phase for bidding
+        require(auctions[slot].appliedAccountStateIndex == 0, "Auction already applied");
+        require(slot != MAX_UINT && slot <= auctionIndex, "Requested auction slot does not exist");
+
+        uint biddingStartTime = auctions[slot].creationTimestamp + 3 minutes;
+        if (slot > 0 && auctions[slot-1].solutionAcceptedTime > biddingStartTime) {
+            biddingStartTime = auctions[slot-1].solutionAcceptedTime;
+        }
+
+        require(
+            block.timestamp > biddingStartTime || auctions[slot].size == maxUnreservedOrderCount(),
+            "Requested auction slot is still active"
+        );
+        require(
+            block.timestamp < auctions[slot].creationTimestamp + 6 minutes,
```

Comment:
This should be biddingStartTime + 3 minutes, shouldn't it?

Reply:
Yes sir it should!